src/net_processing.cpp

```diff
@@ -731,6 +735,37 @@ std::chrono::microseconds CalculateTxGetDataTime(const uint256& txid, std::chron
     return process_time;
 }
 
+static void RetryProcessTx(CConnman& connman, const uint256& txid, const std::chrono::microseconds current_time) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
+{
+    CNodeState::TxDownloadState* best_d = nullptr;
+    std::chrono::microseconds best;
+
+    for (auto& el : mapNodeState) {
+         CNodeState::TxDownloadState* d = &el.second.m_tx_download;
+         if (d->m_tx_in_flight.size() >= MAX_PEER_TX_IN_FLIGHT) continue;
+         auto it = d->m_tx_announced.find(txid);
+         if (it != d->m_tx_announced.end()) {
+             if (best_d == nullptr || (it->second != std::chrono::microseconds::zero() && it->second < best)) {
+                 best_d = d;
+                 best = it->second;
+             }
+         }
+    }
+
+    std::chrono::microseconds process_time = current_time + GetRandMicros(MAX_NOTFOUND_RETRY_RANDOM_DELAY);
+    if (best_d != nullptr && process_time < best) {
+        auto end = best_d->m_tx_process_time.end();
+        for (auto it = best_d->m_tx_process_time.lower_bound(best); it != end && it->first == best; ++it) {
+            if (it->second == txid) {
+                best_d->m_tx_process_time.erase(it);
```

Comment:
Maybe I'm misunderstanding the logic here, but this will bring the `NodeState`'s `m_tx_process_time` forward (to now + ~1second), but does nothing to the global `g_already_asked_for` time, which means that next time we go round `SendMessages()`, we won't actually rerequest the transaction, since `last_request_time > current_time - GETDATA_TX_INTERVAL`. Am I missing something?

If I'm right, I think we need to do something like reset `g_already_asked_for` to one minute ago.